metricflow/sql/optimizer/tag_column_aliases.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+import logging
+from collections import defaultdict
+from typing import Dict, FrozenSet, Iterable, Mapping, Set
+
+from typing_extensions import override
+
+from metricflow.sql.sql_plan import (
+    SqlCreateTableAsNode,
+    SqlCteNode,
+    SqlQueryPlanNode,
+    SqlQueryPlanNodeVisitor,
+    SqlSelectQueryFromClauseNode,
+    SqlSelectStatementNode,
+    SqlTableNode,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TaggedColumnAliasSet:
```

Comment:
For this class specifically - it feels like the name implies a simple dataclass / storage object. I would recommend changing the name to something like `ColumnAliasCollector` or `SqlNodeColumnAliasLinker`.

Reply:
I went back and forth on the naming of this class - this used to be called a `RequiredColumnAliasCollector` (variable name from the old name here that I forgot to update https://github.com/dbt-labs/metricflow/pull/1501/commits/d01cbeac247c8166b26a2dbf94d727d6f1df57a4#diff-9501bfacb03f699851b322c091e05df399fa2131819f87b48923d4a0ea902206R67) but I renamed it because it is more like a container.

Thinking about some alternatives, but regarding "tag" - what did you think it meant?

Reply:
For me, the word `tag` was pretty meaningless without more context - I had no inclination of what some of these classes/variables did without reading the code. So it wasn't that I thought they did something else, I just had to do a lot of reading to make a guess. If there was more context included in the name it might be more intuitive - something like `SqlNodesTaggedWithColumnAliases` but that's pretty verbose.

I do think that once you read and understand the code thoroughly it the naming makes more sense. But I would love for it to be more intuitive for other devs to skim this, too. (Especially when a DAG is involved, reading the code can take a while because you need to jump between files and IDEs aren't set up to easily click through from function to function.)

I actually thought naming it like a container was misleading. A name like `TaggedColumnAliasSet` sounds to me like a simple frozen dataclass. But this is a container for something very changeable and it also handles operations for traversing a graph and collecting aliases, which I wouldn't guess from the name.